packages/contracts/contracts/L2/teleportr/TeleportrDisbursement.sol

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.9;
+
+contract TeleportrDisburser {
+    address private owner;
+    uint256 public totalDisbursements;
+
+    constructor() {
+        owner = msg.sender;
+        totalDisbursements = 0;
+    }
+
+    struct Disbursement {
+        uint256 amount;
+        address addr;
+    }
+
+    function disburse(uint256 nextDepositId, Disbursement[] calldata _disbursements)
+        external
+        payable
+        onlyOwner
+    {
+        uint256 _numDisbursements = _disbursements.length;
+        require(_numDisbursements > 0, "No disbursements");
+
+        uint256 _depositId = totalDisbursements;
+        require(_depositId + 1 == nextDepositId, "Unexpected next deposit id");
+        totalDisbursements += _numDisbursements;
+
+        uint256 _totalDisbursed = 0;
+        for (uint256 i = 0; i < _numDisbursements; i++) {
+            _totalDisbursed += _disbursements[i].amount;
+        }
+        require(
+            _totalDisbursed == msg.value,
+            "Total to disburse is not equal to amount sent"
+        );
+
+        for (uint256 i = 0; i < _numDisbursements; i++) {
+            _depositId += 1;
+            uint256 _amount = _disbursements[i].amount;
+            address _addr = _disbursements[i].addr;
+            (bool sent, ) = _addr.call{value: _amount, gas: 2300}("");
+            if (sent) emit DisbursementSuccess(_depositId, _addr, _amount);
+            else emit DisbursementFailed(_depositId, _addr, _amount);
+        }
+    }
+
+    modifier onlyOwner() {
+        require(msg.sender == owner, "only accessible to owner");
+        _;
+    }
```

Comment:
Also needs `setOwner()` and `getOwner()` methods like the other contract.

Reply:
made `owner` public and added `setOwner` 👍